### PR TITLE
Have `make dist` cleanup a few more wayward files before tarring

### PR DESCRIPTION
**Pull request by timwoj**

### CODE REVIEW

**Comment by 0xxon** [comment]
Note that this is not quite enough yet, to catch everything. I think in the last cases it was ._* files (aka resource forks) that caused the problems.

**Comment by timwoj** [comment]
I should probably implement the same in all of the submodules as well.

**Comment by timwoj** [comment]
> Note that this is not quite enough yet, to catch everything. I think in the last cases it was ._* files (aka resource forks) that caused the problems.

Apparently macOS `tar` has a way to avoid including those files natively. I updated the commit to use that method instead. I tested this on a Big Sur system, sending the tar files to an Ubuntu 20 system, and the resource files are all gone now.

File: Makefile [hunk 1]
```diff
@@ -32,12 +32,17 @@ docclean:
 livehtml:
 	$(MAKE) -C doc $@
 
+# The first dist target here exports a variable to the environment that keeps tar
+# from including the resource files on macOS.
+dist:   export COPYFILE_DISABLE = true
```
Comment (rsmmr) [inline Makefile:37]: (Wow, that's an odd features; doesn't even seem documented for their `tar`.)

File: Makefile [hunk 2]
```diff
@@ -32,12 +32,17 @@ docclean:
 livehtml:
 	$(MAKE) -C doc $@
 
+# The first dist target here exports a variable to the environment that keeps tar
```
Comment (rsmmr) [inline Makefile:35]: What does it mean to have the same target twice? Is that well-formed and/or maybe a GNUism?

Comment (timwoj) [reply Makefile:35]: Yah, looks like `export` might be a GNUism. That fails on FreeBSD. I guess I could just move setting the variable down before the two tar statements.

File: Makefile (hunk 1 above)
Comment (timwoj) [reply Makefile:37]: Apple's `tar` used to have an option called `--disable-copyfile` that you could pass, but they broke it in the last few versions of macOS.

Comment (timwoj) [reply Makefile:37]: And yah, it's completely undocumented (and unfortunately means it's prone to later breakage). I pulled it out of https://superuser.com/a/260264.

**Comment by timwoj** [comment]
Any other comments on this?

File: Makefile (hunk 2 above)
Comment (rsmmr) [reply Makefile:35]: I'd say pull it down before the tars, then at least if clear what's happening.

File: Makefile (hunk 1 above)
Comment (rsmmr) [reply Makefile:37]: Yeah, that's fine, I was just expressing my amazement. :) Would suggest to put that link into a comment though.

**Review by rsmmr — APPROVED** [review]

**Pull request merged**